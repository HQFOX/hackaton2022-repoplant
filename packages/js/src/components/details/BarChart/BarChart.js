import React, { useEffect, useState } from "react";
import { useTranslation } from "react-i18next";
import { HvBarchart } from "@hv/uikit-react-core/dist";
import { fetchSalesData } from "lib/api/data";

const BarChart = () => {
  const [data, setData] = useState([]);
  const { t } = useTranslation();

  useEffect(() => {
    const fetchData = async () => {
      const result = await fetchSalesData();
      setData(result.data);
    };
    fetchData();
  }, []);

  if (!data || !data.length) return null;

  return (
    <HvBarchart
      title={t("components.details.chart.title")}
      subtitle={t("components.details.chart.subtitle")}
      data={data}
      xAxisTitle={t("components.details.chart.xAxisTitle")}
      yAxisTitle={t("components.details.chart.yAxisTitle")}
    />
  );
};

BarChart.propTypes = {};

export default BarChart;
