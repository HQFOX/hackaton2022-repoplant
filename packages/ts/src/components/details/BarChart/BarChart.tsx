import React, { useEffect, useState } from "react";
import { useTranslation } from "react-i18next";
import { HvBarchart } from "@hv/uikit-react-core/dist";
import { fetchSalesData } from "lib/api/data";

const BarChart: React.FC = () => {
  const [data, setData] = useState([]);
  const { t } = useTranslation();

  useEffect(() => {
    const fetchData = async () => {
      const result = await fetchSalesData();
      // @ts-ignore
      setData(result.data);
    };
    fetchData();
  }, []);

  if (!data || !data.length) return null;

  return (
    <HvBarchart
      title={t("components.details.chart.title")}
      subtitle={t("components.details.chart.subtitle")}
      data={data}
      xAxisTitle={t("components.details.chart.xAxisTitle")}
      yAxisTitle={t("components.details.chart.yAxisTitle")}
    />
  );
};

export default BarChart;
