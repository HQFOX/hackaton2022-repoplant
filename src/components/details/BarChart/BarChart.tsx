import React, { useEffect } from "react";
import { useTranslation } from "react-i18next";
import { HvBarchart } from "@hv/uikit-react-core";

const BarChart = ({ data, getBarData }) => {
  const { t } = useTranslation();

  useEffect(() => {
    getBarData();
  }, [getBarData]);

  return data.length ? (
    <HvBarchart
      title={t("components.details.chart.title")}
      subtitle={t("components.details.chart.subtitle")}
      data={data}
      xAxisTitle={t("components.details.chart.xAxisTitle")}
      yAxisTitle={t("components.details.chart.yAxisTitle")}
    />
  ) : null;
};

export default BarChart;
