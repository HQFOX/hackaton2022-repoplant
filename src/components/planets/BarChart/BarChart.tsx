import React, { useEffect } from "react";
import { useTranslation } from "react-i18next";
import { HvBarchart } from "@hv/uikit-react-core";

const BarChart = ({ data, getBarData }) => {
  const { t } = useTranslation();

  useEffect(() => {
    getBarData();
  }, [getBarData]);

  return data.length ? (
    <HvBarchart
      title={t("components.planets.chart.title")}
      subtitle={t("components.planets.chart.subtitle")}
      data={data}
      xAxisTitle={t("components.planets.chart.xAxisTitle")}
      yAxisTitle={t("components.planets.chart.yAxisTitle")}
    />
  ) : null;
};

export default BarChart;
