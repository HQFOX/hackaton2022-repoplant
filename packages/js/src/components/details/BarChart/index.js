import { withStyles } from "@material-ui/core";
import styles from "./styles";
import BarChart from "./BarChart";

export default withStyles(styles, { name: "BarChart" })(BarChart);
