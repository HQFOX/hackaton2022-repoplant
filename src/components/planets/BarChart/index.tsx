import { bindActionCreators, Dispatch } from "redux";
import { connect } from "react-redux";
import { withStyles } from "@material-ui/core";
import { AppState } from "typings/state";
import { getBarData } from "store/data/thunks";
import styles from "./styles";
import BarChart from "./BarChart";

const mapStateToProps = ({ data }: AppState) => ({
  data: data.bar,
});

const mapDispatchToProps = (dispatch: Dispatch) =>
  bindActionCreators(
    {
      getBarData,
    },
    dispatch
  );

export default withStyles(styles, { name: "BarChart" })(
  connect(mapStateToProps, mapDispatchToProps)(BarChart)
);
