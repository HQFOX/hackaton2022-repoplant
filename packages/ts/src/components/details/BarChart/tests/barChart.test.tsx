import React from "react";
import { mount } from "enzyme";
import BarChart from "..";

describe("<BarChart />", () => {
  let component;

  beforeEach(() => {
    component = mount(<BarChart />);
  });

  it("should be defined", () => {
    expect(component).toBeDefined();
  });

  it("matches the snapshot", () => {
    expect(component).toMatchSnapshot();
  });

  it("should be rendered", () => {
    const barChart = component.find(BarChart);
    expect(barChart.length).toBe(1);
  });
});
