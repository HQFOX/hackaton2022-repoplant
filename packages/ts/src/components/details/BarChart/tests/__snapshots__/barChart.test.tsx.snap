// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BarChart /> matches the snapshot 1`] = `
<WithStyles(BarChart)>
  <BarChart
    classes={Object {}}
  />
</WithStyles(BarChart)>
`;
